mmeds/tests/unit/test.py

```diff
@@ -0,0 +1,144 @@
+                           'single_barcodes',
+                           fig.TEST_READS,
+                           None,
+                           fig.TEST_BARCODES,
+                           fig.TEST_CODE_SHORT))
+        if 'tools' in tests:
+            test_setup.append((fig.TEST_SUBJECT_SHORT,
+                               fig.TEST_SPECIMEN_SHORT,
+                               fig.TEST_DIR,
+                               fig.TEST_USER,
+                               'Test_Paired',
+                               'paired_end',
+                               'single_barcodes',
+                               fig.TEST_READS,
+                               fig.TEST_REV_READS,
+                               fig.TEST_BARCODES,
+                               fig.TEST_CODE_PAIRED))
+            test_setup.append((fig.TEST_SUBJECT_SHORT,
+                               fig.TEST_SPECIMEN_SHORT,
+                               fig.TEST_DIR,
+                               fig.TEST_USER,
+                               'Test_Demuxed',
+                               'single_end',
+                               'single_barcodes',
+                               fig.TEST_DEMUXED,
+                               None,
+                               fig.TEST_BARCODES,
+                               fig.TEST_CODE_DEMUX))
+            test_otu = (fig.TEST_SUBJECT_SHORT,
+                        fig.TEST_SPECIMEN_SHORT,
+                        fig.TEST_DIR,
+                        fig.TEST_USER,
+                        'Test_SparCC',
+                        fig.TEST_OTU,
+                        fig.TEST_CODE_OTU)
+            assert 0 == upload_otu(test_otu)
+    if 'database' in tests:
+        test_setup.append((fig.TEST_SUBJECT,
+                           fig.TEST_SPECIMEN,
+                           fig.TEST_DIR,
+                           fig.TEST_USER,
+                           'Test_Single',
+                           'single_end',
+                           'single_barcodes',
+                           fig.TEST_READS,
+                           None,
+                           fig.TEST_BARCODES,
+                           fig.TEST_CODE))
+        test_setup.append((fig.TEST_SUBJECT_ALT,
+                           fig.TEST_SPECIMEN_ALT,
+                           fig.TEST_DIR_0,
+                           fig.TEST_USER_0,
+                           'Test_Single_0',
+                           'single_end',
+                           'single_barcodes',
+                           fig.TEST_READS,
+                           None,
+                           fig.TEST_BARCODES,
+                           fig.TEST_CODE + '0'))
+    for setup in test_setup:
+        assert 0 == upload_metadata(setup)
+
+def run_tests(tests):
+    test_class = []
+    for test in tests:
+        test_class.append(test.capitalize() + 'Test')
+    test_directory = os.path.dirname(os.path.abspath(__file__))
```

Comment:
Use pathlib to do this for consistency with the rest of MMEDS. os.path is a holdover from python 2